integration/securitypolicy_test.go

```diff
@@ -311,38 +420,41 @@ func TestSecurityPolicy(t *testing.T) {
 					},
 				}
 
-				_, err = util.CreateOrUpdate(ctx, polRec.Client, &policyCR, func() error {
-					return nil
-				})
-				eval.NoErr(err)
-
-				err = wait.For(
-					conditions.New(c.Client().Resources()).ResourceMatch(&policyCR, func(object k8s.Object) bool {
-						_, err = polRec.Reconcile(ctx, ctrl.Request{NamespacedName: cr.ObjectKeyFromObject(&policyCR)})
-						return err == nil
-					}),
-					wait.WithTimeout(defaultWaitTimeout),
-					wait.WithInterval(defaultWaitInterval),
-				)
+				// Create the SecurityPolicy on k8s.
+				err = c.Client().Resources(testNs).Create(ctx, &policyCR)
 				eval.NoErr(err)
 
 				err = wait.For(
 					conditions.New(c.Client().Resources()).ResourceMatch(&policyCR, func(object k8s.Object) bool {
-						pol, ok := object.(*v1alpha1.SecurityPolicy)
+						policyOnK8s, ok := object.(*v1alpha1.SecurityPolicy)
 						eval.True(ok)
 
-						eval.True(len(pol.Status.PolID) > 0)
-						eval.True(pol.Status.PolID == policyCR.Spec.MID)
-						eval.Equal(len(pol.Spec.AccessRightsArray), 1)
-
 						// Ensure that policy is created on Tyk
-						policyOnTyk, err := klient.Universal.Portal().Policy().Get(reqCtx, pol.Status.PolID)
-						eval.NoErr(err)
+						policyOnTyk, err := klient.Universal.Portal().Policy().Get(reqCtx, policyOnK8s.Status.PolID)
+						if err != nil {
+							t.Logf("Failed to find Policy %v on Tyk, err: %v", policyOnK8s.Status.PolID, err)
+							return false
+						}
 
-						eval.Equal(pol.Status.PolID, policyOnTyk.MID)
-						eval.Equal(pol.Spec.Name, policyOnTyk.Name)
-						eval.Equal(len(pol.Spec.AccessRightsArray), len(policyOnTyk.AccessRightsArray))
-						eval.Equal(pol.Spec.AccessRightsArray[0].APIID, policyOnTyk.AccessRightsArray[0].APIID)
+						eval.True(len(policyOnK8s.Status.PolID) > 0)
```

Comment:
This is not required now because you are getting the policy using this ID before this check.
If it is empty, policy get will fail.